alvr/xtask/flatpak/audio-flatpak-setup.sh

```diff
@@ -19,52 +31,52 @@ function get_alvr_playback_source_id() {
-  # This sink is required so that it persistently auto-connects to alvr playback later
-  pactl load-module module-null-sink sink_name=ALVR-MIC-Sink media.class=Audio/Sink
-  # This source is required so that any app can use it as microphone
-  pactl load-module module-null-sink sink_name=ALVR-MIC-Source media.class=Audio/Source/Virtual
-  # We link them together
-  pw-link ALVR-MIC-Sink ALVR-MIC-Source
-  # And we assign playback of pipewire alsa playback to created alvr sink
-  pactl move-sink-input "$(get_alvr_playback_source_id sink-inputs 'Sink Input' 'ALSA plug-in [vrserver]')" "$(get_sink_id ALVR-MIC-Sink)"
+  if [[ $USE_HEADSET_MIC == 1 ]]; then
+    echo "Creating microphone sink & source and linking alvr playback to it"
+    # This sink is required so that it persistently auto-connects to alvr playback later
+    pactl load-module module-null-sink sink_name=ALVR-MIC-Sink media.class=Audio/Sink | tee -a /run/user/1000/alvr-audio
+    # This source is required so that any app can use it as microphone
+    pactl load-module module-null-sink sink_name=ALVR-MIC-Source media.class=Audio/Source/Virtual | tee -a /run/user/1000/alvr-audio
+    # We link them together
+    pw-link ALVR-MIC-Sink:monitor_FL ALVR-MIC-Source:input_FL
+    pw-link ALVR-MIC-Sink:monitor_FR ALVR-MIC-Source:input_FR
+    # And we assign playback of pipewire alsa playback to created alvr sink
+    pactl move-sink-input "$(get_playback_sink_input_id 'ALSA plug-in [vrserver]')" "$(get_sink_id_by_name ALVR-MIC-Sink)"
+    pactl set-default-source ALVR-MIC-Source
+  fi
 }
 
-function setup_audio() {
-  echo "Setting up audio"
-  pactl load-module module-null-sink sink_name=ALVR-AUDIO-Sink media.class=Audio/Sink
-  pactl set-default-sink ALVR-AUDIO-Sink
-  pactl move-source-output "$(get_alvr_playback_source_id source-outputs 'Source Output' 'ALSA plug-in [vrserver]')" "$(get_sink_id ALVR-AUDIO-Sink)"
+function unload_modules() {
+  echo "Unloading audio, microphone sink & source"
+  while read -r line; do
+    pactl unload-module "$line"
+  done <"/run/user/1000/alvr-audio"
+  >/run/user/1000/alvr-audio
 }
 
-function unload_mic() {
-  echo "Unloading microphone sink & source"
-  pw-cli destroy ALVR-MIC-Sink
-  pw-cli destroy ALVR-MIC-Source
-}
-
-function unload_sink() {
-  echo "Unloading audio sink"
-  pw-cli destroy ALVR-AUDIO-Sink
+function setup_audio() {
+  if [[ $USE_HEADSET_AUDIO == 1 ]]; then
+    echo "Setting up audio"
+    pactl load-module module-null-sink sink_name=ALVR-AUDIO-Sink media.class=Audio/Sink | tee -a /run/user/1000/alvr-audio
+    pactl set-default-sink ALVR-AUDIO-Sink
+    pactl move-source-output "$(get_playback_source_output_id 'ALSA plug-in [vrserver]')" "$(get_sink_id_by_name ALVR-AUDIO-Sink)"
+  fi
 }
 
 case $ACTION in
 connect)
-  unload_sink
-  unload_mic
-  sleep 1
-  setup_mic
+  unload_modules
   setup_audio
+  setup_mic
   ;;
 disconnect)
-  unload_mic
-  unload_sink
+  unload_modules
   ;;
-esac
+esac
```

Comment:
tiny nit but this should follow LF, not CRLF

Reply:
it was LF though, just no line after it

Reply:
added line after it